networks/dyn_img_primal_dual_nn.py

```diff
@@ -0,0 +1,183 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import numpy as np
+from einops import rearrange
+
+from .grad_ops import GradOperators
+from .prox_ops import ClipAct
+
+
+class DynamicImagePrimalDualNN(nn.Module):
+    def __init__(
+        self,
+        T=128,
+        cnn_block=None,
+        mode="lambda_cnn",
+        up_bound=0,
+        phase="training",
+    ):
+        super(DynamicImagePrimalDualNN, self).__init__()
+
+        # gradient operators and clipping function
+        dim = 3
+        self.GradOps = GradOperators(dim, mode="forward", padmode="circular")
+
+        # operator norms
+        self.op_norm_AHA = torch.sqrt(
+            torch.tensor(1.0)
+        )  # op-norm is one for appropriate csms
+        self.op_norm_GHG = torch.sqrt(torch.tensor(12.0))
+
+        # operator norm of K = [A, \nabla]
```

Comment:
delete comment
"op-norm is one for appropriate csms"
as it is only makes sense for the MRI example